modules/apps/frontend-js/frontend-js-state-web/src/main/resources/META-INF/resources/State.ts

```diff
@@ -514,11 +515,8 @@ window.Liferay.State = State;
  */
 
 export default State;
-
 declare global {
-	interface Window {
-		Liferay: {
-			State: typeof State;
-		};
+	export interface ILiferay {
+		State?: typeof State;
```

Comment:
How does this work? Is it merged with `SPA` in https://github.com/liferay-frontend/liferay-portal/pull/1379/files#diff-10174dbf5a248993084321a6662596ff7faa394aed5696d0fd001030e3544a0bR140 ?

Reply:
Yeah, it'll get merged with whatever global declaration of the interface is there. The thing is you need to make sure everyone contributing those types is loaded all the time which is not the case currently... need to figure out a different approach, likely involving `@types` and `typeRoots` or sth like that... 🤔

Reply:
Of course, even better would be if we stopped using the `Liferay` global object 😂

Reply:
Of course, of course,... :sweat_smile: